test/shmemx/c11_shmemx_team_reduce.c

```diff
@@ -0,0 +1,260 @@
+      case or:                                                                 \
+          CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                                   \
+          break;                                                               \
+      case xor:                                                                \
+          CHECK_DEST_BUFFER(OP, TYPE, (TYPE)(npes % 2 ? 1ULL : 0ULL));         \
+          break;                                                               \
+      case max:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      case min:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      case sum:                                                                \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, npes, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, npes);                               \
+          break;                                                               \
+      case prod:                                                               \
+          if (floating_point_val)                                              \
+              CHECK_DEST_BUFFER_FP(OP, TYPE, 1ULL, FLOATING_POINT_TOLERANCE);  \
+          else                                                                 \
+              CHECK_DEST_BUFFER(OP, TYPE, 1ULL);                               \
+          break;                                                               \
+      default:                                                                 \
+          printf("Invalid operation (%d)\n", OP);                              \
+          shmem_global_exit(1);                                                \
+    }                                                                          \
+  } while (0)
+
+#else
+#define TEST_SHMEM_REDUCE(OP, TYPE)
+#endif
+
+
+int main(void) {
+
+    shmem_init();
+
+    int rc = EXIT_SUCCESS;
```

Comment:
`rc` seems to be unnecessary as it is never used other than returning success.

Reply:
`rc` can be set to `EXIT_FAILURE` as a side effect of the macros that test individual operations.  It might be nice to pass `rc` to the macros rather than taking it as an implicit argument, but it would also add clutter, making an already complex code more difficult to read.  I think I am ok with this test as-is.